qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -13,13 +13,16 @@
 
 """Synthesize higher-level objects."""
 
+from typing import Union, List
 
 from qiskit.converters import circuit_to_dag
 from qiskit.transpiler.basepasses import TransformationPass
 from qiskit.dagcircuit.dagcircuit import DAGCircuit
 from qiskit.transpiler.exceptions import TranspilerError
-from qiskit.synthesis import synth_clifford_full
-from qiskit.synthesis.linear import synth_cnot_count_full_pmh
+from qiskit.transpiler.coupling import CouplingMap
+from qiskit.synthesis.clifford import synth_clifford_full
+from qiskit.synthesis.linear import synth_cnot_count_full_pmh, synth_cnot_depth_line_kms
+from qiskit.synthesis.linear.linear_circuits_utils import optimize_cx_4_options, _compare_circuits
```

Comment:
I wonder if we should change `optimize_cx_4_options` to `_optimize_cx_4_options` ?

Reply:
I don't feel strongly one way or another, but since you are mentioning this, it does feel that this function is rather "internal" than "external". As it does not appear in the previous release notes, I am indeed going to rename it.